prow/sidecar/run.go

```diff
@@ -208,17 +222,34 @@ func (o Options) preUpload() {
 	}
 }
 
-func (o Options) doUpload(ctx context.Context, spec *downwardapi.JobSpec, passed, aborted bool, metadata map[string]interface{}, logReaders map[string]io.Reader) error {
+func (o Options) doUpload(ctx context.Context, spec *downwardapi.JobSpec, passed, aborted bool, metadata map[string]interface{}, logReaders map[string]io.Reader, logFile *os.File, once *sync.Once) error {
 	startTime := time.Now()
 	logrus.Info("Starting to upload")
-	defer func() { logrus.WithField("duration", time.Since(startTime).String()).Info("Finished uploading") }()
-
 	uploadTargets := make(map[string]gcs.UploadFunc)
 
+	defer func() {
+		logrus.WithField("duration", time.Since(startTime).String()).Info("Finished uploading")
+	}()
+
 	for logName, reader := range logReaders {
 		uploadTargets[logName] = gcs.DataUpload(reader)
 	}
 
+	logFileName := logFile.Name()
+
+	once.Do(func() {
+		logrus.SetOutput(os.Stderr)
+		logFile.Sync()
+		logFile.Close()
+	})
+
+	f, err := os.Open(logFileName)
+	if err != nil {
+		logrus.Fatalf("Could not open log file: %v", err)
```

Comment:
Please use `.WithError()` rather than a format string.
Also we should log this at the error level and put lines 250-251 in an `else` block so that this isn't a fatal error.